runtime/cudaq/distributed/mpi_plugin.cpp

```diff
@@ -92,6 +92,17 @@ void MPIPlugin::broadcast(std::vector<double> &data, int rootRank) {
       m_comm, data.data(), data.size(), FLOAT_64, rootRank));
 }
 
+void MPIPlugin::broadcast(std::string &data, int rootRank) {
+  std::int32_t strLen = data.size();
+  HANDLE_MPI_ERROR(
+      m_distributedInterface->Bcast(m_comm, &strLen, 1, INT_32, rootRank));
+
+  if (rank() != rootRank)
+    data.resize(strLen);
```

Comment:
This seems redundant, but maybe the call to `Bcast` is changing the size of the string object... :shrug:

Reply:
Indeed, we need to allocate these string buffers before using MPI API to populate them.
Generally, only `rootRank` knows the size of the string, hence we need to do a two-pass `Bcast` as above.